src/scripts/main.js

```diff
@@ -1 +1,21 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const mainImage = document.querySelector('#largeImg');
+
+  const thumbnails = document.querySelector('#thumbs');
+
+  function changeImage(e) {
+    const clickedImage = e.target.tagName === 'IMG' ? e.target : null;
+    const necessaryLink = clickedImage ? clickedImage.closest('a') : null;
+
+    if (clickedImage && necessaryLink) {
+      e.preventDefault();
+
+      const newSrc = necessaryLink.getAttribute('href');
+
+      mainImage.setAttribute('src', newSrc);
+    }
```

Comment:
This block of code will only execute if the clicked element is an image and it's inside a link. But if the user clicks on the link but outside the image, this code won't execute, and the main image won't change. To fix this, you should first find the closest link to the clicked element, then check if it contains an image, and if it does, change the main image's source to the link's href value.